src/add-to-project.ts

```diff
@@ -157,6 +181,76 @@ export async function addToProject(): Promise<void> {
     )
 
     core.setOutput('itemId', addResp.addProjectV2DraftIssue.projectItem.id)
+    itemId = addResp.addProjectV2DraftIssue.projectItem.id
+  }
+
+  
+  // Next, use the GraphQL API to request the project's fields using the node ID.
+  const fieldsResp = await octokit.graphql<ProjectFieldsResponse>(
+    `query getProjectFields($projectId: ID!) {
+      node(id: $projectId) {
+        ... on ProjectV2 {
+          fields(first: 50) {
+            nodes {
+              ... on ProjectV2Field {
+                id
+                dataType
+                name
+              }
+              ... on ProjectV2SingleSelectField {
+                id
+                name
+                dataType
+                options {
+                  id
+                  name
+                }
+              }
+            }
+          }
+        }
+      }
+    }`,
+    {
+      projectId,
+    },
+  )
+    
+  if (!fields.length) {
+    return
+  }
```

Comment:
This should go above `fieldsResp` - we don't need this request if there are not fields.